### PR TITLE
439 Update sync script module flag

### LINKED ISSUE #439: Update sync script module flag
The -m module argument is currently needed to specify the module name in the results syncing script. We should document that behavior, or instead update the script to make it a required argument sans flag.

### PR DESCRIPTION
**Pull request by jashapiro**
Closes #439

While the modules flag was already required, I decided that for simplicity we should just make it a positional argument, so that is what I did here.

I updated error messages as well, and the docs to match the new format.

Finally, I removed a script that we do not need that was meant to be part of the openscpca-nf repo.

### CODE REVIEW

File: scripts/sync-results.py [hunk 1]
```diff
@@ -27,10 +27,9 @@ def main() -> None:
         description="Sync data from an analysis module to your S3 bucket.",
     )
     parser.add_argument(
-        "--module",
-        "-m",
+        "module",
+        metavar="MODULE",
```
Comment (sjspielman) [inline scripts/sync-results.py:31]: noticed while I was here - maybe change the help below to "The name of the analysis module to sync **RESULTS** from.", (not data)

**Review by sjspielman — APPROVED** [review]
LGTM! Script behaves as expected.

I left 1 small comment on code not formally in this PR.

File: scripts/sync-results.py (hunk 1 above)
Comment (sjspielman) [reply scripts/sync-results.py:31]: hmmm... "to sync results _for_"? either way.

**Review by allyhawkins — APPROVED** [review]
I would incorporate Stephanie's suggestion but otherwise LGTM

File: scripts/sync-results.py [hunk 2]
```diff
@@ -27,10 +27,9 @@ def main() -> None:
...
         type=str,
-        required=True,
         help="The name of the analysis module to sync data from.",
```
Comment (jashapiro) [inline scripts/sync-results.py:33]:
```suggestion
        help="The name of the analysis module to sync results from.",
```

**Review by sjspielman — APPROVED** [review]
actually approving!

**Pull request merged**